app/grandchallenge/components/backends/docker.py

```diff
@@ -252,11 +254,19 @@ def _provision_input_volume(self):
             self._copy_input_files(writer=writer)
 
     def _copy_input_files(self, writer):
-        for file in self._input_files:
+        for input_file in self._input_files:
+            if isinstance(input_file, tuple):
+                name, input_file = input_file
+                if not hasattr(input_file, "name"):
+                    file = NamedTemporaryFile(delete=True)
+                    file.write(bytes(str(input_file), "utf-8"))
+                    file.flush()
+                    input_file = File(file, name=name)
+            else:
+                name = input_file.name
+            self._create_subdirs(os.path.join("/input", *name.split("/")[:-1]))
```

Comment:
```suggestion
            subdirs = os.path.join("/input", *name.split("/")[:-1])
            writer.exec_run(f"mkdir -p {subdirs}")
```

Then `_create_subdirs` can be removed, saves orchestrating another container instance for each file